index.bikeshed.bs

```diff
@@ -343,7 +361,55 @@ implementation detail. It is not an API that implementors
 provide to web applications. It is used in this document
 only to simplify the algorithm description.
 
-## Response verification algorithms ## {#verification-algorithms}
+## Request verification algorithms ## {#request-verification-algorithms}
+
+### Opting-in  ### {#opt-in-require-sri-for}
+
+Authors may opt a {{Document}} to require SRI metadata be present for 
+some resource types via a <dfn export>require-sri-for</dfn> <a>Content
+Security Policy</a> directive defined by the following ABNF grammar:
+
+<pre dfn-type="grammar" link-type="grammar">
+    directive-name  = "require-sri-for"
+    directive-value = <a grammar>token</a> *( <a>RWS</a> <a>token</a> )
+</pre>
+
+The directive recognizes a number of potential token values:
+
+  * `script` requires SRI for scripts
+  * `style` requires SRI for style sheets
+
+### Parsing `require-sri-for` ### {#parse-require-sri-for}
+
+To parse the |token| list, the user agent MUST use an algorithm equivalent to the following:
+
+1.  Let the set of |protected resource types| that require SRI be the empty set.
+
+2.  For each token returned by <a>splitting tokens on spaces</a>,
+  if token matches the grammar for <a>require-sri-for</a>,
+  add the token to the set of |protected resource types|. Otherwise, ignore the token.
+
+3.  Return the set of |protected resource types|.
+
+### Apply |algorithm| to |request| ### {#apply-algorithm-to-request}
+
+This directive’s <a>pre-request check</a> is as follows:
+
+Given a <a>request</a> (|request|) and a <a>policy</a> (|policy|):
+
+1.  Let |protected resource types| be an empty set.
+
+2.  If a directive whose name is "require-sri-for" is present in policy’s directive set,
```

Comment:
This will only be called when such a directive is present. You can rephrase this a bit by replacing 1 and 2 with something like:

```
1.  Let |protected resource types| be the result of executing [[#parse-require-sri-for]] on this directive's <a for="directive">value</a>.
```